### PR TITLE
Student Upload form Added

### PR DESCRIPTION
**Pull request by srivastavaritik**

### CODE REVIEW

File: src/components/home/student/AddStudent.js [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+      }, 3000);
+    }
+  };
+  const config = {
+    headers: {
+      Authorization: "Token " + localStorage.getItem("token"),
+    },
+  };
+ useEffect(() => {
+   axios.get(`http://localhost:8000/api/faculty/subjects/`, config).then((res) => {
```
Comment (jindal2209) [inline src/components/home/student/AddStudent.js:95]: use http_lib
add urls to constants

take latest pull of master and merge inside your branch

Comment (srivastavaritik) [reply src/components/home/student/AddStudent.js:95]: ok

File: src/components/home/chart.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import Title from './title';
 function createData(time, amount) {
   return { time, amount };
 }
-
+//dummy-data
```
Comment (jindal2209) [inline src/components/home/chart.js:10]: remove this

File: src/components/home/student/AddStudent.js [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+
+  useEffect(() => {
+    http_lib.get(COURSE_URL).then((res) => {
+      const data = res.data;
+      setCourseList(data);
+    });
+  }, []);
+  useEffect(() => {
+    console.log(courseList);
+  }, [courseList]);
```
Comment (jindal2209) [inline src/components/home/student/AddStudent.js:103]: no need to log

File: src/components/home/student/AddStudent.js [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+  useEffect(() => {
```
Comment (jindal2209) [inline src/components/home/student/AddStudent.js:104]: write inside the above useEffect

File: src/components/home/student/AddStudent.js [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+    http_lib.get(BRANCH_URL).then((res) => {
+      const data = res.data;
+      setBranchList(data);
+    });
+  }, []);
+  useEffect(() => {
```
Comment (jindal2209) [inline src/components/home/student/AddStudent.js:110]: no need if we are not doing anything inside

File: src/components/home/student/AddStudent.js [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+    console.log(branchList);
+  }, [branchList]);
+
+  const [filesToUpload, setFilesToUpload] = useState([]);
```
Comment (jindal2209) [inline src/components/home/student/AddStudent.js:114]: initialise everything at top

File: src/components/home/student/AddStudent.js [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+    // Update chosen files
+    setFilesToUpload([...files]);
+  };
+
+  const uploadFiles = () => {
+    // Create a form and post it to server
+    let formData = new FormData();
+    filesToUpload.forEach((file) => formData.append("files", file));
+
+    http_lib("/file/upload", {
```
Comment (jindal2209) [inline src/components/home/student/AddStudent.js:126]: is this backend api?

File: src/components/home/student/AddStudent.js [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+  const [student_phone_number, setStudent_phone_number] = useState("");
+  const [student_email_id, setStudent_email_id] = useState("");
+  const [mother_name, setMother_name] = useState("");
+  const [father_name, setFather_name] = useState("");
+  const [parent_phone_number, setParent_phone_number] = useState("");
+  const [branch, setBranch] = useState();
+  const [errorText, setErrorText] = useState("");
+  const [successText, setSuccessText] = useState("");
+
+  const data = {
```
Comment (jindal2209) [inline src/components/home/student/AddStudent.js:35]: whats the use of doing this?

File: src/components/home/student/AddStudent.js [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+import FileUpload from "react-mui-fileuploader";
+import { BRANCH_URL, COURSE_URL } from "../../../constants/api_endpoints";
+import http_lib from "../../../http/http";
+import "../style.css";
+
+const AddStudent = () => {
+  const [courseList, setCourseList] = useState([]);
+  const [branchList, setBranchList] = useState([]);
+  const [course, setCourse] = useState("");
+  const [enrollment_number, setEnrollment_number] = useState("");
```
Comment (jindal2209) [inline src/components/home/student/AddStudent.js:20]: pls use camelCase

Comment (jindal2209) [reply src/components/home/student/AddStudent.js:20]: ref: https://reactjs.org/docs/dom-elements.html

Comment (jindal2209) [reply src/components/home/student/AddStudent.js:20]: instead use formik for form handling
ref: https://formik.org/

File: src/components/home/student/AddStudent.js (hunk 7 above)
Comment (srivastavaritik) [reply src/components/home/student/AddStudent.js:126]: no, its just dummy. I will change it when I'll implement

File: src/components/home/student/AddStudent.js (hunk 8 above)
Comment (jindal2209) [reply src/components/home/student/AddStudent.js:35]: cleanup

**Pull request closed without merging**